### PR TITLE
Design document

### PR DESCRIPTION
**Pull request by LoganTHarvell**
Feel free to edit as you see fit before merging.

### CODE REVIEW

File: Documentation/Game Design.txt [hunk 1]
```diff
@@ -16,43 +16,77 @@ UI:
...
     - Cursor selects planets
     - Arrow keys control ship
+
   => Current planet info
-    - Displays info when a ship collides with a planet on the map
+    - Displays planet info when a ship collides with a planet on the map
+
   => Selected planet info
-    - Displays info when a player hovers the cursor over a planet
+    - Displays planet info when a player hovers the cursor over a planet
```
Comment (holdemani) [inline Documentation/Game Design.txt:24]: Do we want it to be hovers over, or clicks? I feel like hovering would be annoying

File: Documentation/Game Design.txt [hunk 2]
```diff
@@ -16,43 +16,77 @@ UI:
...
 
 Gameplay:
   => Exploration
-    - Move ship around the map to visit each planet
-    - Movement on the map consumes fuel
+    - Planets
+      > Galaxy contains fixed number of planets
+      > Planets can be undiscovered, discovered, colonized (state machine)
+        + Undiscovered planets can't be selected
+        + Discovered planets become colonized if population is > 0, vice versa
```
Comment (holdemani) [inline Documentation/Game Design.txt:35]: Vice versa? You mean it becomes uncolonized if population == 0?

Comment (holdemani) [reply Documentation/Game Design.txt:35]: discovered* my bad

File: Documentation/Game Design.txt [hunk 3]
```diff
@@ -16,43 +16,77 @@ UI:
...
+  => Alien Warship
+    - Alien warship roams and can potentially wipe planet population
 
   => Endgame:
-    - Discover all planets
+    - Maximizing population gives score
+    - Discover all planets, "win condition" bonus?
     - Run out of fuel in space
     - Run out of fuel on planet with no deposits
+    - Ship collides with Alien Warship
```
Comment (holdemani) [inline Documentation/Game Design.txt:69]: lol rip

File: Documentation/Game Design.txt [hunk 4]
```diff
@@ -16,43 +16,77 @@ UI:
...
-    - Can be colonized by sending population to surface
-      > Population can be used to gather resources
-    - Events/disasters can happen periodically
+  => Alien Warship
+    - Alien warship roams and can potentially wipe planet population
 
   => Endgame:
-    - Discover all planets
+    - Maximizing population gives score
+    - Discover all planets, "win condition" bonus?
```
Comment (holdemani) [inline Documentation/Game Design.txt:66]: I mean, if they're gonna discover all the planets, they're already going to have a higher score than if they didn't. So why need a bonus? Unless it's used to entice them not to lose

File: Documentation/Game Design.txt [hunk 5]
```diff
@@ -16,43 +16,77 @@ UI:
...
     - Run out of fuel in space
     - Run out of fuel on planet with no deposits
+    - Ship collides with Alien Warship
 
 AI:
+  => Map generation
+    - homeworld vs planet, resources
```
Comment (holdemani) [inline Documentation/Game Design.txt:73]: Why did you use "vs"? That gives it a competition vibe

Comment (holdemani) [reply Documentation/Game Design.txt:73]: I also think the first planet should be plentiful (enough to get going) in resources and be immune to events for the first X amount of time

File: Documentation/Game Design.txt [hunk 6]
```diff
@@ -16,43 +16,77 @@ UI:
...
+
+  => Population Growth
+
+  => Random events
+    - fair/balanced randomness
```
Comment (holdemani) [inline Documentation/Game Design.txt:78]: Are there good events as well? Or just bad events like your evil famine?

File: Documentation/Classes.txt [hunk 7]
```diff
@@ -1,20 +1,78 @@
...
-    - struct Planet
-      > enum Status
+    - Fields
+      > Galaxy
+    - Methods
+      > Handle Events
+      > Update
+      > Render
+
+  => Galaxy (maybe pointless?)
```
Comment (holdemani) [inline Documentation/Classes.txt:11]: lmao in all honesty, it's only purpose is to make the planet array, and we pass that into map, so why not just have map do it all?

**Review by holdemani — CHANGES_REQUESTED** [review]

File: Documentation/Game Design.txt (hunk 1 above)
Comment (LoganTHarvell) [reply Documentation/Game Design.txt:24]: Probably clicks, but we can always change our minds later

File: Documentation/Game Design.txt (hunk 2 above)
Comment (LoganTHarvell) [reply Documentation/Game Design.txt:35]: Yes

File: Documentation/Game Design.txt (hunk 4 above)
Comment (LoganTHarvell) [reply Documentation/Game Design.txt:66]: Well discovering planets doesn't add to their score except by enabling a higher population, which they may or may not take the time to do. The idea is yes, to make them want to discover all the planets to "win". This adds to the challenge and gives motivation to discover every planet

File: Documentation/Game Design.txt (hunk 5 above)
Comment (LoganTHarvell) [reply Documentation/Game Design.txt:73]: This is under A.I. and Map generation. Given the context obviously Homeworld vs Planet is not competition, its just the A.I. generating the planets intelligently, distinguishing the Homeworld from the rest of the galaxy. Your second comment is why there's a distinction between the initialization of the Homeworld and the rest of the planets

File: Documentation/Game Design.txt (hunk 6 above)
Comment (LoganTHarvell) [reply Documentation/Game Design.txt:78]: Just bad events so far, but we can add good events if it seems fun. However, birth rates can be high for a period of time which is sort of a good "event"

File: Documentation/Classes.txt (hunk 7 above)
Comment (LoganTHarvell) [reply Documentation/Classes.txt:11]: You do not want Map to contain the planets. It has nothing to do with anything but UI and rendering. We may end up just instantiating a planets array called galaxy in Game, but it is also possible we will replace the galaxy class with a GalaxyManager that constructed by passing an array of planets and the galaxy manager then has methods to handle events and update the planets by calling planet methods i.e. getters and setters for deposits, minerals, population

**Comment by LoganTHarvell** [comment]
If there's any further changes you would like to make after my replies, feel free to edit them or discuss them. However, I think it can be merged for now, feel free to leave the branch and we will update further

File: Documentation/Game Design.txt (hunk 1 above)
Comment (holdemani) [reply Documentation/Game Design.txt:24]: For testing we can start with hovering, but I wanna change that before completion

File: Documentation/Game Design.txt (hunk 6 above)
Comment (holdemani) [reply Documentation/Game Design.txt:78]: oh it's good alright ;)

**Review by holdemani — APPROVED** [review]

**Pull request merged**